cmd/govim/quickfix.go

```diff
@@ -44,6 +44,12 @@ func (v *vimstate) updateQuickfixWithDiagnostics(force bool, wasPrevNotDiagnosti
 	if (!force && !work) || !v.quickfixIsDiagnostics {
 		return nil
 	}
+	var qflist qflistProps
+	v.Parse(v.ChannelExpr(`getqflist({"title":1})`), &qflist)
+	if !force && qflist.Title != "" && qflist.Title != "govim" {
```

Comment:
On thing I would like to improve is to merge this condition and the one from `v.quickfixIsDiagnostics`.

The later is set to false when Govim populates the quickfix with references, but we could use quickfix title also in that case.

I think if I turn `v.quickfixIsDiagnostics` into a function that returns true if the quickfix title is empty or equal to "govim" (maybe rename that to "govim-diagnostics", because we could add an other title for Govim references), then it would clean a little bit the code :

- only one condition here instead of 2
- no more manual `v.quickfixIsDiagnostics = true|false` since it relies on quickfix title

Reply:
Yes I'd be happy to revisit this. Either in this PR or a later one. As you observed, it's rather clumsy now. Vim is essentially the source of truth for what is in the quickfix window at any given time, and your change helps move us closer to that arrangement.